src/runtime_src/core/edge/drm/zocl/zocl_xclbin.c

```diff
@@ -453,27 +463,40 @@ zocl_xclbin_load_pdi(struct drm_zocl_dev *zdev, void *data)
 	}
 
 	mutex_lock(&zdev->zdev_xclbin_lock);
-	/* Check unique ID */
-	if (zocl_xclbin_same_uuid(zdev, &axlf_head->m_header.uuid)) {
-		DRM_INFO("%s The XCLBIN already loaded, uuid: %pUb",
-			 __func__, &axlf_head->m_header.uuid);
-		mutex_unlock(&zdev->zdev_xclbin_lock);
-		return ret;
-	}
-
-	write_lock(&zdev->attr_rwlock);
 
-	/* Get full axlf header */
 	size_of_header = sizeof(struct axlf_section_header);
 	num_of_sections = axlf_head->m_header.m_numSections-1;
 	xclbin = (char __user *)axlf;
-	ret =
-	    !ZOCL_ACCESS_OK(VERIFY_READ, xclbin, axlf_head->m_header.m_length);
+
+	ret = !ZOCL_ACCESS_OK(VERIFY_READ, xclbin, axlf_head->m_header.m_length);
 	if (ret) {
 		ret = -EFAULT;
 		goto out;
 	}
 
+	write_lock(&zdev->attr_rwlock);
+
+	zocl_read_sect(AIE_RESOURCES, &aie_res, axlf, xclbin);
+	if (is_aie_only(axlf)) {
+		ret = zocl_load_aie_only_pdi(zdev, axlf, xclbin, NULL);
+		if (ret) {
+			DRM_WARN("read xclbin: fail to load AIE");
+			goto out;
+		}
+		size = zocl_read_sect(AIE_METADATA, &zdev->aie_data.data, axlf, xclbin);
+		if (size > 0 ) {
+			zdev->aie_data.size = size;
+			zocl_create_aie(zdev, axlf, aie_res);
+		}
```

Comment:
Minor thing: we probably don't need to check if there is AIE_METADAT here. It will be checked inside ocl_create_aie so that we have a centralize place to do the sanity check.
However, more important, zocl_create_aie was added and tested for edge flow only so far. We may need to do the same thing for PCIe as well but I suggest we do it in a separate PR where we have a clear use case for zocl to request AIE partition to support control AIE from PS kernel and related support. This PR is for load/reload AIE PDI only, right?